apps/session-manager/src/controllers/__tests__/zendesk.spec.ts

```diff
@@ -0,0 +1,244 @@
+/* eslint-disable max-lines-per-function */
+import { describe, it, vi, expect, beforeEach } from "vitest";
+import {
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseErrorTooManyRequests,
+  ResponseSuccessJson,
+} from "@pagopa/ts-commons/lib/responses";
+import * as express from "express";
+import * as TE from "fp-ts/TaskEither";
+import * as E from "fp-ts/Either";
+import { pipe } from "fp-ts/lib/function";
+import { Second } from "@pagopa/ts-commons/lib/units";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import mockReq from "../../__mocks__/request.mocks";
+import { ZendeskController } from "../";
+import {
+  mockedInitializedProfile,
+  mockedUser,
+} from "../../__mocks__/user.mocks";
+import mockRes from "../../__mocks__/response.mocks";
+import { FnAppAPIClient } from "../../repositories/fn-app-api";
+import { mockRedisClientSelector } from "../../__mocks__/redis.mocks";
+import { ProfileService, TokenService } from "../../services";
+
+const mockGetProfile = vi.spyOn(ProfileService, "getProfile");
+
+const mockGetZendeskSupportToken = vi.spyOn(
+  TokenService,
+  "getJwtZendeskSupportToken",
+);
+
+const aZendeskSupportToken =
+  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiTWFyaW8gUm9zc2kiLCJlbWFpbCI6Im1hcmlvLnJvc3NpQHRlc3QuaXQiLCJleHRlcm5hbF9pZCI6IkFBQUFBQUFBQUFBQUFBQSIsImlhdCI6MTYzNTk0NDg2MC43ODksImp0aSI6IjAxRktKWUszM04wRkpKNEVZRTM5NURIWlhDIiwiZXhwIjoxNjM1OTQ0OTIwLCJpc3MiOiJJU1NVRVIifQ.dqQlnKGME5FvpI2GbkVpk7vpgE_ft0IZE3jp2YRWHtA";
+
+describe("ZendeskController#getZendeskSupportToken", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+  const res = mockRes() as unknown as express.Response;
+  const req = mockReq() as unknown as express.Request;
+
+  const mockedDependencies = {
+    // Repositories are not used, since we mocked the service layer
+    fnAppAPIClient: {} as ReturnType<typeof FnAppAPIClient>,
+    redisClientSelector: mockRedisClientSelector,
+    user: mockedUser,
+    req,
+    jwtZendeskSupportTokenSecret: "aSecret" as NonEmptyString,
+    jwtZendeskSupportTokenExpiration: 1200 as Second,
+    jwtZendeskSupportTokenIssuer: "issuer" as NonEmptyString,
+  };
+
+  it("should return a valid Zendesk support token when user has a validated email", async () => {
+    mockGetProfile.mockReturnValue(
```

Comment:
It's safer to use `mockReturnValueOnce`, for providing better test isolation.

Reply:
changed in https://github.com/pagopa/io-auth-n-identity-domain/pull/34/commits/15b5283dc8882ae73e35c2dc05e3f91746718016